Flow.Launcher/Storage/UserSelectedRecord.cs

```diff
@@ -8,16 +10,42 @@ namespace Flow.Launcher.Storage
     public class UserSelectedRecord
     {
         [JsonInclude]
-        public Dictionary<string, int> records { get; private set; }
+        public Dictionary<int, int> records { get; private set; }
 
         public UserSelectedRecord()
         {
-            records = new Dictionary<string, int>();
+            records = new Dictionary<int, int>();
+        }
+
+        private static int GenerateCustomHashCode(Query query, Result result)
+        {
+            int hashcode = 23;
+
+            unchecked
+            {
+                // skip the empty space
+                foreach (var item in query.ActionKeyword.Concat(query.Search))
+                {
+                    hashcode = hashcode * 31 + item;
+                }
+
+                foreach (var item in result.Title)
+                {
+                    hashcode = hashcode * 31 + item;
+                }
+
+                foreach (var item in result.SubTitle)
+                {
+                    hashcode = hashcode * 31 + item;
```

Comment:
31?

Reply:
checkout https://stackoverflow.com/a/5155015
31 will have a better performance without information loss when int overflow

Reply:
Yeah, let's give it a bit of description, set it as a descriptive variable and reference that article so in the future we know where and why the logic is constructed